packages/financial-templates-lib/src/price-feed/UniswapPriceFeed.js

```diff
@@ -227,15 +244,47 @@ class UniswapPriceFeed extends PriceFeedInterface {
       return reserve1.mul(fixedPointAdjustment).div(reserve0);
     }
   }
+}
 
-  _computeTwap(eventsIn, startTime, endTime) {
-    const events = eventsIn.map(e => {
-      return [e.timestamp, e.price];
+class UniswapV3PriceFeed extends UniswapPriceFeed {
```

Comment:
this is super clean. I like that it's also within the same file as most of the logic is re-used. very nice implementation!